### PR TITLE
[GHA] Cache cleanup workload

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Introduced the cleanup workload to remove all generated caches for specific PR when it closed
 - And remove old master caches expect the latest one

### Tickets:
 - *n/a*

### CODE REVIEW

File: .github/workflows/cleanup_cache.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: write
+      contents: read
+
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      REPO: ${{ github.repository }}
+      BRANCH: refs/pull/${{ github.event.pull_request.number }}/merge
```
Comment (akashchi) [inline .github/workflows/cleanup_cache.yml:18]: I suppose that `${{ github.ref }}` would be the right one?
image

File: .github/workflows/cleanup_cache.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            echo "${CACHE_KEYS_OLD[@]}"
+            set +e
+            if [[ ${#CACHE_KEYS_OLD[@]} -gt 0 ]]
+            then
+              echo "Deleting caches..."
+              for CACHE_KEY in ${CACHE_KEYS_OLD[@]}
+              do
+                if [[ ! -z "$CACHE_KEY" ]]
+                then
+                  gh actions-cache delete $CACHE_KEY -R $REPO -B $BRANCH --confirm
```
Comment (akashchi) [inline .github/workflows/cleanup_cache.yml:58]: I suggest adding some logging, e.g.: `echo "DELETING CACHE KEY: $CACHE_KEY"` just for the easier possible debug.

File: .github/workflows/cleanup_cache.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Cleanup master's caches
+        if: github.event.pull_request.merged == true
+        env:
+          BRANCH: master
+          KEYS: "ccache-linux-ubuntu ccache-Build-linux-debian ccache-Build-linux-android ccache-Build-linux-arm64"
+        run: |
+          for KEY in ${KEYS[@]}
+          do
+            echo "Fetching list of cache with key ${KEY} for ${BRANCH}"
+            CACHE_KEYS=( $(gh actions-cache list -R $REPO -B $BRANCH -L 100 --key $KEY --sort created-at --order desc | cut -f 1) )
```
Comment (akashchi) [inline .github/workflows/cleanup_cache.yml:47]: Enclosed in `()` for readability?

File: .github/workflows/cleanup_cache.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          do
+            gh actions-cache delete $CACHE_KEY -R $REPO -B $BRANCH --confirm
+          done
+          echo "Done"
+
+      - name: Cleanup master's caches
+        if: github.event.pull_request.merged == true
+        env:
+          BRANCH: master
+          KEYS: "ccache-linux-ubuntu ccache-Build-linux-debian ccache-Build-linux-android ccache-Build-linux-arm64"
```
Comment (akashchi) [inline .github/workflows/cleanup_cache.yml:42]: Do they find the cache entries with timestamps in them?

File: .github/workflows/cleanup_cache.yml (hunk 3 above)
Comment (mryzhov) [reply .github/workflows/cleanup_cache.yml:47]: no to initialize variable as array

File: .github/workflows/cleanup_cache.yml (hunk 4 above)
Comment (mryzhov) [reply .github/workflows/cleanup_cache.yml:42]: yes, it finds all entries including entries with timestamps

**Pull request closed without merging**